test/functional/query-builder/cte/simple-cte.ts

```diff
@@ -0,0 +1,114 @@
+import "reflect-metadata";
+import { expect } from "chai";
+import { createTestingConnections, closeTestingConnections, reloadTestingDatabases } from "../../../utils/test-utils";
+import { Connection } from "../../../../src/connection/Connection";
+import { Foo } from "./entity/foo";
+
+
+describe("query builder > cte > simple", () => {
+
+
+    let connections: Connection[];
+    before(async () => connections = await createTestingConnections({
+        entities: [__dirname + "/entity/*{.js,.ts}"],
+        schemaCreate: true,
+        dropSchema: true,
+        enabledDrivers: [
+            "postgres",
+            "mariadb",
+            "sqlite",
+            "mssql",
+            // "mysql", // Version 8.0 is required for CTEs in MySQL
```

Comment:
should test Oracle and CockroachDB as well

Reply:
Done.